opacus/privacy_engine.py

```diff
@@ -41,9 +49,57 @@ def make_private(
             max_grad_norm=max_grad_norm,
             expected_batch_size=expected_batch_size,
             loss_reduction=loss_reduction,
+            distributed=distributed,
+        )
+
+        def accountant_hook(optim: DPOptimizer):
+            # TODO: This works for Poisson for both single-node and distributed
+            # The reason is that the sample rate is the same in both cases (but in distributed mode, each node samples among a subset of the data)
+            self.accountant.step(
+                noise_multiplier=optim.noise_multiplier,
+                sample_rate=sample_rate * optim.accumulated_iterations,
+            )
+
+        optimizer.attach_step_hook(accountant_hook)
+
+        return module, optimizer, data_loader
+
+    def make_private_per_layer(
```

Comment:
@ffuuugor Could you remind what we concluded on this on that meeting? 

How I interpret the idea:
```python
# Here I draw an analogy with sklearn's Estimator instance
privacy_engine = PrivacyEngine(
    accountant="rdp",
    clipping="flat",
    poisson=True,
    noise_multiplier=1.0,
)

# Here I draw an analogy with sklearn's 
#  output = estimator.transform(input)
model, optim, dl = privacy_engine.make_private(model, optim, dl)
```

Here we maintain separation between hyperparameters and the object to which we apply the method. This pattern is common in ML libs, I think we should follow it too. Also, it can make [Lightning integration] cleaner.

Reply:
We want to keep things like `noise_multiplier` and `max_grad_norm` in `make_private` to allow dynamic changes to these parameters during training (and that would be properly reflected in accounting).
We have also already committed to a certain API in a DevDay presentation. 

I think we should be open to changing the API for compelling reason, but so far it looks to me that moving stuff to the constructor instead of make_private will hurt the functionality